contrib/package_plugin.py

```diff
@@ -0,0 +1,313 @@
+    with tempfile.TemporaryDirectory() as temp_path:
+        # Export the generated 'manifest.json' file into place.
+        manifest_file_path = os.path.join(temp_path, "manifest.json")
+        write_manifest(metadata, manifest_file_path)
+
+        # Copy the selected Python package into place.
+        dest_package_path = os.path.join(temp_path, package_directory_name)
+        shutil.copytree(source_package_path,  dest_package_path)
+        # Python bytecode cannot be written into the zip archive as Electron Cash runs it.
+        # So we precompile it before creating the archived form.
+        compileall.compile_dir(dest_package_path)           
+        shutil.make_archive(suffixless_path, 'zip', temp_path)
+        
+    plugin_path = suffixless_path +".zip"
+    hash_md5 = hashlib.md5()
+    with open(plugin_path, "rb") as f:
+        hash_md5.update(f.read())
+    return hash_md5.hexdigest()
+
+def write_manifest(metadata, manifest_file_path):
+    with open(manifest_file_path, "w") as f:
+        json.dump(metadata, f, indent=4)
+
+def build_manifest(display_name, version, description, minimum_ec_version, package_name, available_for_qt, available_for_cmdline):
+    metadata = {}
+    metadata["display_name"] = display_name
+    version_value = version.strip()
+    try:
+        versiontuple(version_value)
+    except ValueError:
+        version_value = "0.0"
+    metadata["version"] = version_value
+    metadata["description"] = description
+    version_value = minimum_ec_version.strip()
+    try:
+        versiontuple(version_value)
+    except ValueError:
+        version_value = "0.0"
+    metadata["minimum_ec_version"] = version_value
+    if package_name is not None:
+        metadata["package_name"] = package_name
+    available_for = []
+    if available_for_qt:
+        available_for.append("qt")
+    if available_for_cmdline:
+        available_for.append("cmdline")
+    metadata["available_for"] = available_for
+    
+    return metadata
+
+# A line edit that can be selected and copied, but not edited.
+class XLineEdit(QLineEdit):
+    def keyPressEvent(self, event):
+        event.ignore()
+
+class App(QWidget):
+    def __init__(self):
+        super().__init__()
+         
+        self.directory_path = None
+         
+        self.setWindowTitle('Electron Cash Plugin Packager')
+        self.setMinimumWidth(400)
+        self.setMaximumWidth(400)
+        
+        outerVLayout = QVBoxLayout()
+        self.setLayout(outerVLayout)
+        
+        self.importButton = QPushButton("Import Existing 'manifest.json'")
+        outerVLayout.addWidget(self.importButton)
+        
+        groupBox = QGroupBox("Metadata")
+        groupLayout = QFormLayout()
+        outerVLayout.addWidget(groupBox)
+        label = QLabel("This is a test")
+        groupBox.setLayout(groupLayout)
+        self.displayNameEdit = QLineEdit()
+        self.displayNameEdit.setPlaceholderText("Scheduled Payments")
```

Comment:
I wanted to use realistic examples as placeholders.  So I just used text from my own plugin.  I'm all for text relating to some made up plugin, rather than just another form what the label should serve to say.  This does highlight that the text should serve as a style guide and a better model of what a description should look like should be used than the clumsy one I put there.